internal/workers_service.go

```diff
@@ -0,0 +1,123 @@
+package myhwproj
+
+import (
+	"encoding/json"
+	"log"
+	"math/rand"
+
+	"github.com/streadway/amqp"
+)
+
+// WorkersService defines the api for service which works with Workers who check Submissions
+type WorkersService interface {
+	SetHandler(func(submission Submission))
+	CheckSubmission(submission Submission)
+}
+
+// Helper function to check rabbitmq errors
+func failOnError(err error, msg string) {
+	if err != nil {
+		log.Fatalf("%s: %s", msg, err)
+	}
+}
+
+// RabbitMQWorkersService works with Workers through RabbitMQ
+type RabbitMQWorkersService struct {
+	ResultHandler func(submission Submission)
+	HomeworkDao   HomeworkDao
+}
+
+// NewRabbitMQWorkersService creates default RabbitMQWorkersService
+func NewRabbitMQWorkersService(homeworkDao HomeworkDao) WorkersService {
+	return &RabbitMQWorkersService{
+		ResultHandler: func(submission Submission) {},
+		HomeworkDao:   homeworkDao,
+	}
+}
+
+// SetHandler changes the handler of check result
+func (r *RabbitMQWorkersService) SetHandler(f func(submission Submission)) {
+	r.ResultHandler = f
+}
+
+type RabbitMQMessage struct {
+	Submission Submission
+	Homework   Homework
+}
+
+func randInt(min int, max int) int {
+	return min + rand.Intn(max-min)
```

Comment:
```suggestion
	return min + rand.Intn(max - min)
```
Я go вообще почти не знаю, так что может у вас так принято, тогда можно не править :)

Reply:
Там вроде code style встроен прямо в язык, вроде он даже не соберется с неправильным форматированием

Reply:
👍